cmd/security-secretstore-setup/secretstore-init.sh

```diff
@@ -0,0 +1,6 @@
+#!/usr/bin/dumb-init /bin/sh
```

Comment:
@anonymouse64 Yes!  In fact, in TingYu's refactoring, the token issuing code has been completely removed because the Kong setup has been changed for the time being to use the root token.  Why do you prefer a go program that forks subprocesses vs a shell script that just does (subprocess & ; subprocess &)
cc: @jim-wang-intel

Reply:
While I do write quite a bit of shell code myself in snapd and in authoring snaps (see all the scripts in edgex-go for example under snap/), shell code is very often brittle and difficult to review. There are all kinds of gotchas and weird bugs with different versions of shells you use and while you solve some of that by sticking to strict POSIX with `/bin/sh`, you lose out on a whole lot of useful constructs that greatly simplify the code you write. Furthermore, the project as a whole has decided on Go as the primary language of development and as such I think we should be consistent. It will be easier for other non-security developers to review, it is more portable to different platforms (i.e. if you had a single go program you don't need to worry about all of these issues with forking and reaping and signals and such in docker containers, and you also get windows support pretty easily and don't need to worry about the differences in shell environments between Mac and Linux)

To summarize:
* Shell code is very brittle and rots very quickly
* Go code is more readable 
* Shell code is difficult to write _always_ correctly without extensive use of something like `shellcheck`
* Go has been adopted by the wider EdgeX community, the security services should follow this pattern
* Go code is more easily ported than shell code

I guess finally, my question to you would be: Why do we need the go program to fork subprocesses? Why can't the go program just do the operations it needs to? I see your proposals on the various issues about splitting up the responsibilities but what I don't see is a justification for why we should split these things up other than it separates responsibilities.

Reply:
@anonymouse64  Primary reason is our discussion earlier about future capabilities for the Vault token server: namely, be able to have different implementations.  One implementation would create vault tokens statically and it would be wired up into the health check for secret store readiness.  The other one we discussed was a domain socket API that could create them on the fly.  Given that go plugins are not a viable option at this time, the only other alternatives we have for that type of configuration are (1) separate processes, or (2) include both options into the same executable.

I think it important to decide now what architectural pattern we are going to use when we want to give users a choice.

Reply:
@anonymouse64 Other similar choices in the architecture that are coming up include how we add Vault master key protection and be able to extend it to hardware solutions.

Reply:
well at the minimum we should open up discussion around switching to shell instead of go to the wider community and not just hash it out in this PR...

Also we already have a pattern for this in the SMA, by providing a configuration.toml specification for what binary to run, that could be used here too. If there are concerns about reaping those processes, I think we should look at concrete issues/bugs there before deciding that forking programs in Go is a non-starter, because AFAICT Go does it correctly...

Reply:
@anonymouse64 That is agreeable to me.  In that case we should switch the dockerfile back to FROM scratch and nix the entrypoint script.

Reply:
switched Dockerfile back to scratch per the discussion